source/Engine.cpp

```diff
@@ -928,6 +932,25 @@ void Engine::Draw() const
 	draw[drawTickTock].Draw();
 	batchDraw[drawTickTock].Draw();
 
+	Point dCenter = player.FlagshipPtr()->Position();
+	for(const shared_ptr<Ship> &ship : ships)
+	{
+		if(ship->Cloaking() && ship->IsYours() && ship->GetSystem() == player.GetSystem())
+		{
+			float cloak = .1f * static_cast<float>(ship->Cloaking());
+			OutlineShader::Draw(ship->GetSprite(), (ship->Position() - dCenter) * zoom, Point(ship->Width(), ship->Height()) * zoom,
+								Color(6 * cloak, cloak, cloak, 8 * cloak), ship->Facing().Unit(), ship->GetFrame());
+		}
+
+		if(Preferences::Has("Damage highlights"))
+		{
+			float shields = min(SHIELD_OUTLINE_MULT * static_cast<float>(sqrt(ship->RecentShieldDamage() / ship->Attributes().Get("shields"))), 1.f);
+			if(ship->RecentShieldDamage() > 4. && ship->GetSystem() == player.GetSystem())
+				OutlineShader::Draw(ship->GetSprite(), (ship->Position() - dCenter) * zoom, Point(ship->Width(), ship->Height()) * zoom,
+									Color(.61f * shields, .78f * shields, shields, shields), ship->Facing().Unit(), ship->GetFrame());
+		}
+	}
```

Comment:
Put the two if statements inside the for loop into their own lambdas.
Then, do:
```c++
if(Preferences::Has("Damage highlights")
    for(<every ship>)
        if(<ship is in player system>)
        {
            CloakLambda(ship);
            ShieldDamageHighlight(ship);
        }
else
    for(<every ship>)
        if(<ship is in player system>)
            CloakLambda(ship);
```